tests/document/instance.py

```diff
@@ -3186,8 +3181,7 @@ class BlogPost(Document):
 
         blog.update(push__tags__0=["mongodb", "code"])
         blog.reload()
-        self.assertEqual(blog.tags[0], "mongodb")
-        self.assertEqual(blog.tags[2], "python")
+        self.assertEqual(blog.tags, ['mongodb', 'code', 'python'])
```

Comment:
Ah, I just noticed you said in one of your comments "position modifier must appear with each operator, then we could supply list value even if single element". Does that mean that you *must* supply a list value to e.g. `push__tags__1`? If `push__tags='single_val'` works, then `push__tags__1='single_val'` should work as well. Do you agree?